### PR TITLE
Issue #4314: Split and Organize Checkstyle inputs by Test for checks in design package: DesignForExtension

### PR DESCRIPTION
**Pull request by Kietzmann**
#4314

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtension.java [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+// Test case file for checkstyle.
+// Created: 2001
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.design.designforextension;
+
+/**
+ * Test case for the "design for inheritance" check.
+ * @author Lars Kühne
+ **/
+public abstract class InputDesignForExtension
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtension.java]: this file is used only ones in this Check only , please move file.

Comment (Kietzmann) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtension.java]: It also used in HideUtilityClassConstructor

Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtension.java]: @Kietzmann Please show us where.
https://github.com/checkstyle/checkstyle/search?utf8=%E2%9C%93&q=InputDesignForExtension&type=

It doesn't show up on a search.

Comment (Kietzmann) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtension.java]: @rnveach @romani  
It was used here under its original name here
[HideUtilityClassConstructorCheck](https://github.com/checkstyle/checkstyle/blob/34d0abd082756bdf9c864821b4225bc21922a0f0/src/test/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheckTest.java#L88)
Before you merged it few hours earlier.
I've fixed this already.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.puppycrawl.tools.checkstyle.checks.design.designforextension;
+
+public class InputDesignForExtensionsLocalAnnotations
+{
+    public @interface Rule {
```
Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java]: @romani Like I asked in a similar PR, should we keep this file just for its public interfaces used in another input, or should we merge it so our inputs are completely standalone?

Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java]: It is better to keep inputs independent on any other file. We do not need good design in this files, the aglier the better.

Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java]: All inputs should be compiled by itself only. The same as we demand from users on when they report issue.

Comment (Kietzmann) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java]: @romani 
Have I understood you correctly that I shouldn't modify this file and all should be as is now?

Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java]: If in original file this class was embedded - keep it as it was

Comment (rnveach) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/design/designforextension/InputDesignForExtensionsLocalAnnotations.java]: Ok, we will leave these for now and get them later.

**Pull request merged**